Add $ to Demos (#5352)
ros2/ros2_documentation: source/Tutorials/Demos/Rosbag-with-ROS1-Bridge.rst

First we'll run a ROS 1 ``roscore`` in a new shell:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

         $ . /opt/ros/kinetic/setup.bash

>          $ roscore

Reviewer: ```suggestion
         $ roscore

```

   .. group-tab:: macOS

      .. code-block:: console

         $ . ~/ros_catkin_ws/install_isolated/setup.bash
         $ rocore

> 

Reviewer: ```suggestion
```

First we'll run a ROS 1 ``roscore`` in a new shell:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

         $ . /opt/ros/kinetic/setup.bash
         $ roscore

> 
> 

Reviewer: Extra line

```suggestion

```

   .. group-tab:: macOS

      .. code-block:: console

         $ . ~/ros_catkin_ws/install_isolated/setup.bash
         $ rocore

Then we'll run the ROS 1 <=> ROS 2 ``dynamic_bridge`` with the ``--bridge-all-topics`` option (so we can do ``rostopic list`` and see them) in another shell:

> .. note::

Reviewer: ```suggestion
.. note::

```

   If you installed rosbridge from source, adapt the path to the setup file accordingly:
   ``. <workspace-with-bridge>/install/setup.bash``.

>    Remember to replace ``<workspace-with-bridge>`` with the path to where you either extracted the ROS 2 binary or where you built ROS 2 from source.

Reviewer: These two sentences are basically saying the same thing

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

       $ . /opt/ros/kinetic/setup.bash
       $ . /opt/ros/ardent/setup.bash
       $ export ROS_MASTER_URI=http://localhost:11311
       $ ros2 run ros1_bridge dynamic_bridge --bridge-all-topics

   .. group-tab:: macOS

      .. code-block:: console

       $ . ~/ros_catkin_ws/install_isolated/setup.bash
       $ . /opt/ros/ardent/setup.bash
       $ export ROS_MASTER_URI=http://localhost:11311

>        $ ros2 run ros1_bridge dynamic_bridge --bridge-all-topics

Reviewer: Even though it seems to work, please indent the code by at least 1 more space so that they're indented by at least 2 spaces

----

Now we can start up the ROS 2 programs that will emulate our turtlebot-like robot.

> First we'll run the ``cam2image`` program with the ``-b`` option so it doesn't require a camera to work, open another shell:

Reviewer: A comma here doesn't really combine these two parts properly; you're better off just having separate sentences:

```suggestion
First we'll run the ``cam2image`` program with the ``-b`` option so it doesn't require a camera to work.
In another shell:
```

You can run this python script in a new ROS 2 shell:

> 

Reviewer: ```suggestion
```

.. code-block:: console

   $ . /opt/ros/ardent/setup.bash
   $ python3 emulate_kobuki_node.py

.. note::

   If building ROS 2 from source adapt the path to the setup file accordingly: ``<workspace-with-bridge>/install/setup.bash``.

----

Now that all the data sources and the dynamic bridge are running, we can look at the available topics in a new ROS 1 shell:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

       $ . /opt/ros/kinetic/setup.bash
       $ rostopic list
       /image
       /imu_data
       /odom
       /rosout
       /rosout_agg

> 
> 

Reviewer: ```suggestion

```

   .. group-tab:: macOS

      .. code-block:: console

       $ . ~/ros_catkin_ws/install_isolated/setup.bash
       $ rostopic list
       /image
       /imu_data
       /odom
       /rosout
       /rosout_agg

> 

Reviewer: Please remove these extra leftover lines so that there's only one line between content

Then in a new shell start the ``roscore``:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

       $ . /opt/ros/kinetic/setup.bash
       $ roscore

   .. group-tab:: macOS

      .. code-block:: console

       $ . ~/ros_catkin_ws/install_isolated/setup.bash
       $ roscore

> 

Reviewer: ```suggestion
```

Then run the ``dynamic_bridge`` in another shell:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

       $ . /opt/ros/kinetic/setup.bash
       $ . /opt/ros/ardent/setup.bash
       $ export ROS_MASTER_URI=http://localhost:11311
       $ ros2 run ros1_bridge dynamic_bridge --bridge-all-topics

   .. group-tab:: macOS

      .. code-block:: console

       $ . ~/ros_catkin_ws/install_isolated/setup.bash
       $ . /opt/ros/ardent/setup.bash
       $ export ROS_MASTER_URI=http://localhost:11311

>        $ ros2 run ros1_bridge dynamic_bridge --bridge-all-topics

Reviewer: Please indent the code by at least 1 more space

Author: Pretty sure that i did this wrong in some other places. Code should be indented by at least two spaces relative to the .. code-block:: right?

> 

Reviewer: ```suggestion
```

Then play the bag data back with ``rosbag play`` in another new shell, using the ``--loop`` option so that we don't have to keep restarting it for short bags:

.. tabs::

   .. group-tab:: Linux

      .. code-block:: console

       $ . /opt/ros/kinetic/setup.bash
       $ rosbag play --loop path/to/bag_file

   .. group-tab:: macOS

      .. code-block:: console

       $ . ~/ros_catkin_ws/install_isolated/setup.bash

>        $ rosbag play --loop path/to/bag_file

Reviewer: Please indent the code by at least 1 more space

> 

Reviewer: ```suggestion
```

> .. note::
>    Make sure to replace ``path/to/bag_file`` with the path to the bag file you want to playback.

Reviewer: Please try to keep these unrelated changes (in this case, making this a `note`) for a separate PR. But for now:

```suggestion
.. note::

   Make sure to replace ``path/to/bag_file`` with the path to the bag file you want to play back.
```

----

Now that the data is being played back and the bridge is running we can see the data coming across in ROS 2.

.. code-block:: console

   $ . /opt/ros/ardent/setup.bash
   $ ros2 topic list
   /clock
   /image
   /imu_data
   /odom
   /parameter_events
   $ ros2 topic echo /odom

> 

Reviewer: ```suggestion
```
